ingester/util.go

```diff
@@ -0,0 +1,39 @@
+package ingester
+
+import (
+	"fmt"
+	"net"
+)
+
+type sortableUint32 []uint32
+
+func (ts sortableUint32) Len() int           { return len(ts) }
+func (ts sortableUint32) Swap(i, j int)      { ts[i], ts[j] = ts[j], ts[i] }
+func (ts sortableUint32) Less(i, j int) bool { return ts[i] < ts[j] }
+
+// getFirstAddressOf returns the first IPv4 address of the supplied interface name.
+func getFirstAddressOf(name string) (string, error) {
```

Comment:
They do different things - alertmangers gets the first non-empty address, this gets the address for a given interface.  I'll file a ticket to reconcile them later.